scripts/apps/workspace/content/components/ContentProfileFieldsConfig.tsx

```diff
@@ -0,0 +1,634 @@
+    return [
+        ...getAllContentProfileSections().map((section) => ({
+            label: getLabelForSection(section),
+            value: section,
+        })),
+        {label: gettext('Widgets'), value: 'widgets'},
+    ];
+}
+
+function getLabelForSection(section: IContentProfileSection) {
+    if (section === IContentProfileSection.header) {
+        return gettext('Header');
+    } else if (section === IContentProfileSection.content) {
+        return gettext('Content');
+    } else {
+        return assertNever(section);
+    }
+}
+
+type IPropsItem = IPropsGenericFormItemComponent<IContentProfileFieldWithSystemId> & IAdditionalProps;
+
+// wrapper is used because sortable HOC considers `index` to be its internal prop and doesn't forward it
+class ItemBase extends React.PureComponent<{wrapper: IPropsItem}> {
+    render() {
+        const {item, page, index, inEditMode} = this.props.wrapper;
+        const {sortingInProgress, setIndexForNewItem, getLabel, availableIds} = this.props.wrapper.additionalProps;
+        const isLast = index === page.getItemsCount() - 1;
+
+        const newFieldSelect = (newItemIndex) => (
+            <select
+                onChange={(event) => {
+                    if (event.target.value === '') {
+                        return; // placeholder selected
+                    }
+
+                    const fieldId = event.target.value;
+
+                    setIndexForNewItem(newItemIndex);
+                    page.openNewItemForm({_id: fieldId});
+                }}
+                style={{width: 100}}
+            >
+                <option value="">{gettext('Add field')}</option>
+                {
+                    availableIds.map(({id, label}) => (
+                        <option key={id} value={id}>{label}</option>
+                    ))
+                }
+            </select>
+        );
+
+        return (
+            <div
+                style={{
+                    position: 'relative',
+
+                    // a clone of the element gets appended to the body when sorting is in progress
+                    // in order to be visible, it has to have a higher zIndex than the modal
+                    zIndex: 1051,
+
+                    border: `1px solid ${inEditMode ? 'red' : 'blue'}`,
+                    marginTop: 10,
+                    marginBottom: 10,
+                    paddingTop: 5,
+                    paddingBottom: 5,
+                    cursor: 'pointer',
+                    userSelect: 'none',
+                }}
+                onClick={(e: any) => {
+                    if (
+                        querySelectorParent(e.target, 'select', {self: true}) == null
+                        && querySelectorParent(e.target, 'button', {self: true}) == null
+                    ) {
+                        page.startEditing(item._id);
+                    }
+                }}
+            >
+                {
+                    !sortingInProgress
```

Comment:
No need for a ternary operator here: `sortingInProgress || (<div />)`

Reply:
True, but ternary operators use the same boolean logic so they have that issue too. I think they are a bit more readable than ternary if you don't use both cases of the ternary operator. Not a big deal tho